opencti-platform/opencti-graphql/src/modules/notification/notification-domain.ts

```diff
@@ -48,22 +49,38 @@ const extractUniqRecipient = async (
   user: AuthUser,
   triggerInput: TriggerDigestAddInput | TriggerLiveAddInput,
   type: TriggerType
-): Promise<{ id: string, name: string }> => {
+): Promise<{ recipient: { id: string, name: string }, entityType: string }> => {
   if (triggerInput.recipients && triggerInput.recipients.length > 0) {
     if (!isUserHasCapability(user, SETTINGS_SET_ACCESSES)) {
       throw ForbiddenAccess();
     }
     if (triggerInput.recipients.length > 1) {
       throw UnsupportedError(`Cannot create ${type} trigger for more than one recipient`);
     }
+    // TODO refacto
     const [recipientId] = triggerInput.recipients;
     const recipientUser = await internalLoadById<BasicStoreEntity>(context, user, recipientId, { type: ENTITY_TYPE_USER });
-    if (!recipientUser) {
+    const recipientGroup = await internalLoadById<BasicStoreEntity>(context, user, recipientId, { type: ENTITY_TYPE_GROUP });
```

Comment:
Suggestion : load the entity by id without the `type` filter, and check if the returned entity `entity_type` is User or Group. The recipient will already contain the entity_type so no need to create another object @marieflorescontact